modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/IGConfigurationDisplayContext.java

```diff
@@ -212,25 +168,31 @@ private void _initFolder() throws PortalException {
 		_folderId = DLFolderConstants.DEFAULT_PARENT_FOLDER_ID;
 		_folderInTrash = false;
 		_folderName = StringPool.BLANK;
+
 		_folderNotFound = false;
 
-		DLPortletInstanceSettings dlPortletInstanceSettings =
-			_igRequestHelper.getDLPortletInstanceSettings();
+		try {
+			_folder = _dlPortletInstanceSettingsHelper.getRootFolder();
+		}
+		catch (Exception exception) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(exception);
+			}
 
-		_folderId = dlPortletInstanceSettings.getRootFolderId();
+			_folderNotFound = true;
 
-		if (_folderId == DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) {
 			return;
 		}
 
-		Folder folder = _getFolder();
+		if (_folder == null) {
+			_folderName = LanguageUtil.get(_httpServletRequest, "home");
 
```

Comment:
It's not really an issue but I found it strange that the same behavior was implemented differently here:
https://github.com/liferay-content-management/liferay-portal/pull/6412/commits/811dcd1169a2eaa5da4fb085e146fe823fe62f4f#diff-62be714b65174cbb32caf9d8f9b24cbacd77fe01ca393618d2aff7b7a83093bfR610-R615